### PR TITLE
use R6 classes to represent relationships between slug and entries

### PR DESCRIPTION
**Pull request by fmichonneau**
```r
g <- GlossaryEntry$new(slug = "plus_one", term = "+1",
                       defn = "A vote in favor of something", lang = "en")
g$add_entry(term = "+1",  defn = "Un vote en faveur de quelque chose", lang = "fr")#
# Slug: plus_one
#  + (en) **+1**: A vote in favor of something
#  + (fr) **+1**: Un vote en faveur de quelque chose
g$export()
## A tibble: 2 x 4
#  slug     language term  definition                        
#  <chr>    <chr>    <chr> <chr>                             
# 1 plus_one en       +1    A vote in favor of something      
# 2 plus_one fr       +1    Un vote en faveur de quelque chose
```

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -12,3 +12,4 @@ _site
 dist
 glossary/glossary.yml
 man
+/Gemfile.lock
```
Comment (gvwilson) [inline .gitignore:15]: why the leading `/`?

File: Gemfile [hunk 2]
```diff
@@ -0,0 +1,2 @@
+source 'https://rubygems.org'
+gem 'github-pages', group: :jekyll_plugins
```
Comment (gvwilson) [inline Gemfile:2]: Sure, go ahead, be all professional 'n' stuff :-)

File: R/entry.R [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+    language =  function() {
+      private$.lang
+    },
+    term = function() {
+      private$.term
+    },
+    definition = function() {
+      private$.defn
+    },
+    print = function() {
```
Comment (gvwilson) [inline R/entry.R:25]: I'm new to R6 - is it usual to provide a way to convert to string as well?

**Comment by fmichonneau** [comment]
there are more to it but here is where this PR is at:

```r
library("glossary")
g <- get_glossary()
g
#> A glossary with 152 entries.
define(c("base_r", "peanuts", "tidyverse"), lang = "en",  glossary = g)
#> base R: The basic functions making up the R language. The base packages can
#>   be found in `src/library` and are not updated outside of R; their version
#>   numbers follow R version numbering. Base packages are installed and loaded
#>   with R, while priority packages are installed with base R but must be loaded
#>   prior to use.
#> See also: tidyverse
#> 
#> Peanuts: An American comic strip by Charles M. Schulz which has inspired the
#>   names of R versions.
#> 
#> Tidyverse: A collection of R packages for operating on tabular data in
#>   consistent ways.
#> 
```

File: R/entry.R (hunk 3 above)
Comment (fmichonneau) [reply R/entry.R]: I'm not sure what you mean here...

File: Gemfile (hunk 2 above)
Comment (fmichonneau) [reply Gemfile:2]: I swear by bundler these days.

File: .gitignore (hunk 1 above)
Comment (fmichonneau) [reply .gitignore:15]: that's what magit did... happy to remove it

**Comment by fmichonneau** [comment]
with #3 merged and the above commits:

```r
library(glossary)
g <- get_glossary()
define(c("plus_one"), lang = c("en", "fr"), glossary = g, show_lang = TRUE)
#> +1: A vote in favor of something. (en)
#> +1: Un vote en faveur de quelque chose. (fr)
#> 
```

**Comment by gvwilson** [comment]
Is this mergeable? thx - G

**Comment by fmichonneau** [comment]
yes, I just pushed a little more work I did on this over the weekend.

**Comment by elletjies** [comment]
Adds @fmichonneau as a contributor for doc, review, question, translation.

**Pull request merged**